x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponse.java

```diff
@@ -190,6 +192,11 @@ public String toString() {
         return Strings.toString(ChunkedToXContent.wrapAsToXContent(this));
     }
 
+    @Override
+    public void close() {
+        Releasables.close(pages.stream().map(p -> (Releasable) () -> p.releaseBlocks()).toArray(Releasable[]::new));
```

Comment:
Discussed in Slack - there's not very many `pages` normally, but this is shorter and avoids allocating an `O(pages)`-sized array too:

```suggestion
        Releasables.close(() -> Iterators.map(pages.iterator(), p -> p::releaseBlocks));
```